src/shims/unix/foreign_items.rs

```diff
@@ -207,6 +208,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 }
             }
 
+            "mmap" => {
+                let [addr, length, prot, flags, fd, offset] = this.check_shim(abi, Abi::C {unwind: false}, link_name, args)?;
+                if let Some(ptr) = this.mmap(addr, length, prot, flags, fd, offset)? {
+                    this.write_pointer(ptr, dest)?;
+                } else {
+                    this.write_null(dest)?;
+                }
```

Comment:
Please make `mmap` return a `Scalar`, and don't have any logic here other than just storing that scalar.

Reply:
`mmap` can return a null pointer. So I think this needs to return `Scalar<Option<Provenance>>` , but then I can't call `write_scalar`, because that function takes `Into<Scalar<Provenance>>`.

Reply:
I can do this by returning a `Pointer<Option<Provenance>>` like I did for `mremap`?

Reply:
Ah I think I figured this out.

Reply:
`Scalar<Option<Provenance>>` is not a thing... but looks like you found `Scalar::from_maybe_pointer` or something like that. :)